hydra-essentials/src/main/java/com/addthis/hydra/common/hash/Murmur3HashFunction.java

```diff
@@ -15,7 +15,24 @@
 
 import com.google.common.hash.Hashing;
 
-final class Murmur3HashFunction implements StringAndByteHashFunction {
+public final class Murmur3HashFunction implements StringAndByteHashFunction {
+    private Murmur3HashFunction() {}
+
+    private static class HolderClass {
+        private static final Murmur3HashFunction instance = new Murmur3HashFunction();
+    }
+
+    public static Murmur3HashFunction getInstance() {
+        return HolderClass.instance;
+    }
+
+    public static int getHash(String input) {
+        return Hashing.murmur3_32().hashUnencodedChars(input).asInt();
+    }
+
+    public static int getHash(byte[] input) {
+        return Hashing.murmur3_32().hashBytes(input).asInt();
+    }
```

Comment:
There are multiple ways for thread safe singleton. I prefer one of the ways which I used since it is known to be fast next to enum, but if you prefer shorter code by using static field initialization, I have no problem to modify it.